### PR TITLE
fix: backport upstream fixes for color chooser dialogs

### LINKED ISSUE #13596: Segmentation fault on color picker close
* Electron Version: 1.8.7
* Operating System (Platform and Version): macOS High Sierra (10.13.5)
* Last known working Electron version: Unsure

**Expected Behavior**

Electron application should not crash when a user closes the color picker in macOS.

**Actual behavior**

Occasionally (not every time) when the end user closes a native color picker (from `<input type="color" />`), the browser process crashes.

Sometimes the termination signal is a `Bus error: 10`:

```
Crashed Thread:        0  CrBrowserMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGBUS)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x00007fff8b4060f0
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Bus error: 10
Termination Reason:    Namespace SIGNAL, Code 0xa
Terminating Process:   exc handler [0]

VM Regions Near 0x7fff8b4060f0:
    unused shlib __DATA    00007fff8b3d4000-00007fff8b404000 [  192K] rw-/rwx SM=COW  system shared lib __DATA not used by this process
--> __DATA                 00007fff8b404000-00007fff8b591000 [ 1588K] rw-/rwx SM=COW  /usr/lib/libobjc.A.dylib
    unused shlib __DATA    00007fff8b591000-00007fff8b59a000 [   36K] rw-/rwx SM=COW  system shared lib __DATA not used by this process

Application Specific Information:
Performing @selector(didChooseColor:) from sender NSColorPanel 0x7f90a462a5d0

Thread 0 Crashed:: CrBrowserMain  Dispatch queue: com.apple.main-thread
0   ???                           	0x00007fff8b4060f0 OBJC_METACLASS_$_NSObject + 0
1   com.github.electron.framework 	0x0000000106c029f4 0x106ab3000 + 1374708
2   com.apple.AppKit              	0x00007fff28ab1a43 -[NSApplication(NSResponder) sendAction:to:from:] + 312
3   com.apple.AppKit              	0x00007fff287defab __61-[NSColorPanel _forceSendAction:notification:firstResponder:]_block_invoke + 100
4   com.apple.AppKit              	0x00007fff287dee15 -[NSColorPanel _withColorSettingDisabled:] + 37
5   com.apple.AppKit              	0x00007fff287def41 -[NSColorPanel _forceSendAction:notification:firstResponder:] + 88
6   com.apple.AppKit              	0x00007fff287df136 -[NSColorPanel setColor:] + 182
...
```

Other times it's a `Segmentation fault: 11`:

```
Crashed Thread:        0  CrBrowserMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       EXC_I386_GPFLT
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

Application Specific Information:
Performing @selector(didChooseColor:) from sender NSColorPanel 0x7fd8d6f2ce60

Thread 0 Crashed:: CrBrowserMain  Dispatch queue: com.apple.main-thread
0   com.github.electron.framework 	0x000000010d9cd760 ColorChooserMac::DidChooseColorInColorPanel(unsigned int) + 16
1   com.github.electron.framework 	0x000000010d9cd9f4 0x10d87e000 + 1374708
2   com.apple.AppKit              	0x00007fff28ab1a43 -[NSApplication(NSResponder) sendAction:to:from:] + 312
3   com.apple.AppKit              	0x00007fff287defab __61-[NSColorPanel _forceSendAction:notification:firstResponder:]_block_invoke + 100
4   com.apple.AppKit              	0x00007fff287dee15 -[NSColorPanel _withColorSettingDisabled:] + 37
5   com.apple.AppKit              	0x00007fff287def41 -[NSColorPanel _forceSendAction:notification:firstResponder:] + 88
6   com.apple.AppKit              	0x00007fff287df136 -[NSColorPanel setColor:] + 182
...
```

**To Reproduce**

The easiest way to reproduce is with the packaged application, which can be downloaded from the [GitHub releases page](https://github.com/mjswensen/themer-gui/releases/tag/v0.18.0).

I have not been able to reproduce the problem in development mode, but here is the information if it's helpful:

```sh
$ git clone git@github.com:mjswensen/themer-gui.git
$ npm install
$ npm start
```

1. Hover over a label to reveal the droplet icon.
  - screen shot 2018-07-09 at 4 42 29 pm
2. Click the droplet icon to open the color picker.
3. Use the eyedropper tool to select a color from elsewhere on the screen.
  - image
4. Click the red dot to close the color picker.
  - image

This is the point the application crashes. (Try a few times if it doesn't happen the first time.)

**Screenshots**

crash report

**Additional Information**

The issue _appears_ to only manifest when the eyedropper is used, specifically when selecting colors from a different application. (I haven't seen a crash when the color picker sliders are used to select a color, for example.)

### PR DESCRIPTION
**Pull request by brenca**
#### Description of Change

Fixes #13596 and replaces an old patch with a better one. Both patches should be removed once we get to a chromium version that includes the two fixes:
https://chromium-review.googlesource.com/c/chromium/src/+/1479332
https://chromium-review.googlesource.com/c/chromium/src/+/1469461

Not sure which version that would be, maybe @nornagon can help me with that.
The Windows fix is there to avoid a `DCHECK`, so I left that out of the release notes, since no user should see a debug build.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Backported a fix for crashing color picker dialogs on MacOS.

### CODE REVIEW

**Comment by deepak1556** [comment]
You can use https://chromiumdash.appspot.com/commits to find out which version a commit has landed in. Also with upstream backports its not necessary to find out and specify the version, our `script/git-import-patches` will take care of it during upgrades.

File: patches/common/chromium/color_chooser_mac.patch [hunk 1]
```diff
@@ -0,0 +1,184 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Benedek Heilig <benecene@gmail.com>
+Date: Thu, 21 Feb 2019 17:04:30 +0000
+Subject: Backports an upstream fix I made to the color chooser dialog on macos.
```
Comment (nornagon) [inline patches/common/chromium/color_chooser_mac.patch]: this Subject becomes the commit message when the commits are applied, please keep it short & useful for `git log`!

File: patches/common/chromium/color_chooser_win.patch [hunk 2]
```diff
@@ -0,0 +1,90 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Benedek Heilig <benecene@gmail.com>
+Date: Thu, 21 Feb 2019 17:16:33 +0000
+Subject: Backports an upstream fix I made to the color chooser dialog on
```
Comment (nornagon) [inline patches/common/chromium/color_chooser_win.patch]: same here

**Comment by brenca** [comment]
I updated the patches, thanks for the guidance @nornagon!

**Comment by deepak1556** [comment]
Should the release notes also mention the DCHECK fix on windows ?

**Comment by nornagon** [comment]
DCHECK generally isn't observable by app devs, so I'd say no.

**Pull request merged**